File: allure2-androidj/src/main/java/com/yq/allure2_androidj/common/utils/PropertiesUtils.java
package com.yq.allure2_androidj.common.utils;

import android.util.Log;


import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

import static com.yq.allure2_androidj.common.resultRW.AllureConstants.TAG;

/**
 * The collection of properties utils methods.
 */
public final class PropertiesUtils {

    private static final String ALLURE_PROPERTIES_FILE = "allure.properties";

    private PropertiesUtils() {
    }

    public static Properties loadAllureProperties() {
        final Properties properties = new Properties();
        if (Objects.nonNull(ClassLoader.getSystemResource(ALLURE_PROPERTIES_FILE))) {
            try (InputStream stream = ClassLoader.getSystemResourceAsStream(ALLURE_PROPERTIES_FILE)) {
                properties.load(stream);
            } catch (IOException e) {
                Log.e(TAG,"Error while reading allure.properties file from classpath: ", e);
            }
        }
        properties.putAll(System.getProperties());
        return properties;
    }

}

File: allure2-android/src/main/java/com/yq/allure2_android/common/utils/PropertiesUtils.kt
package com.yq.allure2_android.common.utils

import android.nfc.Tag
import android.util.Log
import java.io.IOException
import java.util.*

object PropertiesUtils {

    private val ALLURE_PROPERTIES_FILE = "allure.properties"
    private val TAG = "${allureTag}PropertiesUtils"

    fun loadAllureProperties(): Properties {
        val properties = Properties()
        ClassLoader.getSystemResource(ALLURE_PROPERTIES_FILE)?.apply {
            try {
                ClassLoader.getSystemResourceAsStream(ALLURE_PROPERTIES_FILE).use { stream -> properties.load(stream) }
            } catch (e: IOException) {
                Log.e(TAG ,"Error while reading allure.properties file from classpath: %s", e)
            }

        }
        properties.putAll(System.getProperties())
        return properties
    }
}